### PR TITLE
#165413100 Create a function to send an email

### PR DESCRIPTION
**Pull request by AJAkimana**
#### Description
An application has to send an email to the user for different functions such as sending an update of application to the user, sending a user a verification email after signup and resetting password after he forgets

#### Type of change
N/A

#### How Has This Been Tested?
It is tested by testing the `npm run test` script

#### Checklist:
N/A

#### Pivotal tracker story ID
[#165413100](https://www.pivotaltracker.com/n/projects/2326984/stories/165413100)

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -5,3 +5,5 @@ DB_NAME= name of your database
 DB_HOST= name of your database
 DB_PORT= name of your database
 TESTDB_NAME= name of your test database
+APP_EMAIL=noreply@riders-ah-backend.com
+APP_LINK=htts://riders-ah-backend.herokuapp.com
```
Comment (Kabalisa) [inline .env.sample:18]: I suggest removing the actual values of a APP_EMAIL and APP_LINK from your .env.sample file and add just the use of each environment variable. and also add the SENDGRID_API_KEY env variable.

File: helpers/utils/mail-sender.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const mailSender = {
+  send(infos) {
+    const sendEmailPromise = new Promise((resolve, reject) => {
+      // Prepare the link
+      const link = `${process.env.APP_LINK}/v1/auth/verify?t=${infos.token}`;
+      const email = {
+        from: process.env.APP_EMAIL,
+        to: infos.email,
+        subject: 'Email verification from RidersAhBackend',
+        text: `<html>Please click on the following <strong><a href="${link}"> link</a></strong> to verify your email</html>`,
```
Comment (Kabalisa) [inline helpers/utils/mail-sender.js]: you can remove the text key from email object as it is the same as html key. for reusability of this function you can also pass as argument the subject and html of the email as you did for the to key with infos.email. you can also use infos.subject and infos.html on the two keys.

**Review by Kabalisa — CHANGES_REQUESTED** [review]
good job @AJAkimana. you can consider the comments I have  given to modify some part of your code

**Review by Niyitangasam — APPROVED** [review]

File: helpers/utils/mail-sender.js (hunk 2 above)
Comment (AJAkimana) [reply helpers/utils/mail-sender.js]: I have done it

**Review by Kabalisa — APPROVED** [review]
@AJAkimana well done. now you can just update your branch with the latest changes from develop.

File: helpers/index.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
-
+import nodemailer from 'nodemailer';
```
Comment (macphilips) [inline helpers/index.js:4]: Format this file properly

**Review by macphilips — CHANGES_REQUESTED** [review]

File: helpers/index.js (hunk 3 above)
Comment (AJAkimana) [reply helpers/index.js:4]: > Format this file properly

I have done with it

**Pull request merged**